sklearn/feature_selection/tests/test_from_model.py

```diff
@@ -122,7 +122,7 @@ def check_max_n_features(est, X, y):
 
     for n_features in range(max_n_features):
         transformer = SelectFromModel(
-            FixedImportanceEstimator(range(1, max_n_features + 1)),
+            FixedImportanceEstimator([1, 2, 2, 3, 3, 3, 4, 4, 4, 4]),
```

Comment:
Yes, it should. I would go further to suggest that we test it has a consistent policy, i.e. that either it always takes the first features in a tie or the last.